mongodbatlasv2/atlas_client.go

```diff
@@ -91,3 +92,22 @@ func UseUserAgent(userAgent string) ClientModifier {
 		return nil
 	}
 }
+
+// IsError checks if API returned known error type.
+func IsError(err error) (*Error, bool) {
+	var openapiError *GenericOpenAPIError
+	if ok := errors.As(err, &openapiError); !ok {
+		return nil, false
+	}
+	errModel := openapiError.Model()
+	return &errModel, true
+}
```

Comment:
is this method necessary now that `GenericOpenAPIError` always returns the correct struct and there's no need for casting?

Reply:
SDK has 2 levels of errors. Internal errors like io/output errors/networking issues etc. Actual API errors represented by GenericOpenAPIError. We need to handle the boilerplate of casting to GenericOpenAPIError to avoid documentation (and general confusion related to how errors are handled.